earthaccess/search.py

```diff
@@ -71,6 +103,30 @@ def hits(self) -> int:
 
         return int(response.headers["CMR-Hits"])
 
+    def get(self, limit: int = 2000) -> list:
+        """Get all the collections (datasets) that match with our current parameters
+        up to some limit, even if spanning multiple pages.
+
+        ???+ Tip
+            The default page size is 2000, we need to be careful with the request size because all the JSON
+            elements will be loaded into memory. This is more of an issue with granules than collections as
+            they can be potentially millions of them.
+
+        Parameters:
+            limit: The number of results to return
+
+        Returns:
+            query results as a list of `DataCollection` instances.
+        """
+
+        response = get_results(self, limit)
+
+        results = list(
+            DataCollection(collection, self._fields) for collection in response
+        )
+
+        return results
```

Comment:
Might need some ruff formatting for line length:

```suggestion
        return list(
            DataCollection(collection, self._fields) for collection in get_results(self, limit)
        )
```